### DISCUSSION TITLE
1 failed tests of 469 header files

### DISCUSSION THREAD

**Discussion by zsh151** [idaholab/moose#22189] [Q&A Getting Started] [answered]
Hi.
I was trying to build libmesh manually. But when I finally ran `make installcheck`, one test failed. So will this affect the results?
 
`Testing Header solvers/petscdmlibmesh.h ...                     [ FAILED ]`

![image](https://user-images.githubusercontent.com/110072634/192130977-7551fb6d-705f-4d9d-bacd-f2591b9cb337.png)

**Comment by GiudGiud** [comment, marked as answer]
Hello

We dont use the PETSc DM by default in moose or libmesh so no it wont.
Any reason you are installing libmesh manually instead of using conda or using the scripts/update_and_rebuild_libmesh?

Guillaume

**Reply by zsh151** [reply]
I'm trying to install Cardinal，but there is some incompatibility with the moose‘s conda environment.
So I should build  PETSc and libMesh under base environment. But when I ran `./contrib/moose/scripts/update_and_rebuild_libmesh.sh`，the errors appeared at the beginning.

**Reply by GiudGiud** [reply]
which errors did you get?
Did you install PETSc first?

**Reply by zsh151** [reply]
Yes, I have built PETSc successfully.
I don't understand that error, please wait for me to recompile and reply to you.

**Reply by zsh151** [reply]
Sorry，I'm late.
The errors I can see when I ran './contrib/moose/scripts/update_and_rebuild_libmesh.sh' 
![image](https://user-images.githubusercontent.com/110072634/192139589-34cc6552-65be-46d8-ad0d-78729d48479d.png)
![image](https://user-images.githubusercontent.com/110072634/192139609-c7e3866b-a736-4a0b-bcd8-e52788cbbd0f.png)

The full output is shown in the following file.
[output.txt](https://github.com/idaholab/moose/files/9640559/output.txt)

**Reply by zsh151** [reply]
These prompt codes can run smoothly 
`--disable-netcdf-4  --disable-dap`
Now libmesh seems to have been successfully installed.

Thank you for your answering.

Zhang

**Reply by GiudGiud** [reply]
It seems you need to install this z library or zlib on your machine to build netcdf4 (related to exodus output)
Do you have admin access on this machine?

PS:
Please avoid screenshots, they are not searchable by others. And harder to read on some devices.

**Reply by zsh151** [reply]
You are right. There is still something wrong with libmesh. 
And I have admin access. Should I run `sudo ./contrib/moose/scripts/update_and_rebuild_libmesh.sh` instead ？
When I intended to install Cardinal directly , the following error came up.
```
Compiling C++ (in opt mode) /home/en/cardinal/contrib/moose/framework/build/unity_src/timeintegrators_Unity.C...
In file included from /home/en/cardinal/contrib/moose/framework/build/unity_src/outputs_formatters_Unity.C:2:
/home/en/cardinal/contrib/moose/framework/src/outputs/formatters/ExodusFormatter.C:17:10: fatal error: libmesh/exodusII.h: No such file or directory
   17 | #include "libmesh/exodusII.h"
      |          ^~~~~~~~~~~~~~~~~~~~
compilation terminated.
```

**Reply by GiudGiud** [reply]
No you should install zlib then try to install libmesh again

**Reply by zsh151** [reply]
Okay，thank you ，I'm on it.

**Reply by zsh151** [reply]
Exactly as you said, the executable has been built successfully by doing so!
Thanks again for your valuable guidance!
